File: RepoTransBench/target_projects/Java/C#/Meituan-Dianping_walle/public_tests/SignatureNotFoundExceptionPublicTest.cs
using System;
using Xunit;
using MeituanDianpingWalle;

namespace MeituanDianpingWalle.PublicTests
{
    public class SignatureNotFoundExceptionPublicTest
    {
        [Fact]
        public void TestConstructorMessage_Public()
        {
            var ex = new SignatureNotFoundException("another message");
            Assert.Equal("another message", ex.Message);
            Assert.Null(ex.InnerException);
        }

        [Fact]
        public void TestConstructorMessageAndCause_Public()
        {
            Exception cause = new NullReferenceException("public cause");
            var ex = new SignatureNotFoundException("public error", cause);
            Assert.Equal("public error", ex.Message);
            Assert.Equal(cause, ex.InnerException);
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/Meituan-Dianping_walle/public_tests/signaturenotfoundexception_public_test.go
package public_tests

import (
	"errors"
	"testing"
)

type SignatureNotFoundException struct {
	Message string
	Cause   error
}

func NewSignatureNotFoundException(message string) error {
	return &SignatureNotFoundException{Message: message}
}

func NewSignatureNotFoundExceptionWithCause(message string, cause error) error {
	return &SignatureNotFoundException{Message: message, Cause: cause}
}

func (e *SignatureNotFoundException) Error() string {
	return e.Message
}

func (e *SignatureNotFoundException) Unwrap() error {
	return e.Cause
}

func TestSignatureNotFoundExceptionPublic_ConstructorMessage_public(t *testing.T) {
	ex := NewSignatureNotFoundException("another message").(*SignatureNotFoundException)
	if ex.Message != "another message" {
		t.Errorf("expected message 'another message', got '%v'", ex.Message)
	}
	if ex.Cause != nil {
		t.Errorf("expected nil cause, got %v", ex.Cause)
	}
}

func TestSignatureNotFoundExceptionPublic_ConstructorMessageAndCause_public(t *testing.T) {
	cause := errors.New("public cause")
	ex := NewSignatureNotFoundExceptionWithCause("public error", cause).(*SignatureNotFoundException)
	if ex.Message != "public error" {
		t.Errorf("expected message 'public error', got '%v'", ex.Message)
	}
	if ex.Cause != cause {
		t.Errorf("expected cause '%v', got '%v'", cause, ex.Cause)
	}
}